### PR TITLE
chore: migrate Log4j to SLF4j

### PR DESCRIPTION
**Pull request by PenghaiZhang**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

In this PR, 189 files have been changed to use SLF4J. 

Most files have only two changes. One is the import of SLF4J and the other one is using SLF4J `LoggerFactory` to instantiate a logger. 

Some files have more changes. For example, I changed  `logger.fatal`  to `logger.error` because the log level of `fatal` is not implemented in SLF4J. Some files had a logger created but never used so I also dropped this kind of code.

Manual testing was done on my local environment to confirm all logs are generated correctly, but once we have all  CI builds completed, I will check the logs.

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/src/com/tle/freetext/XmlSchemaIndexer.java [hunk 1]
```diff
@@ -80,7 +81,7 @@ public void indexOneValue(
               new ByteArrayInputStream(value.getBytes("UTF-8")); // $NON-NLS-1$
           value = new HTMLFilter(bais).getSummary(16384);
         } catch (Exception e) {
-          LOGGER.warn(e);
+          LOGGER.warn(e.getMessage());
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/freetext/XmlSchemaIndexer.java:84]:
```suggestion
          LOGGER.warn(e.getMessage(), e);
```

File: Source/Plugins/Core/com.equella.reporting/src/com/tle/core/reporting/ReportingServiceImpl.java [hunk 2]
```diff
@@ -212,15 +213,15 @@ public void publish(LogRecord record) {
...
+        logger.error(message);
       } else if (Level.INFO.equals(level)) {
         logger.info(message);
       } else if (Level.WARNING.equals(level)) {
         logger.warn(message);
       } else if (Level.FINE.equals(level)
           || Level.FINER.equals(level)
           || Level.FINEST.equals(level)) {
-        logger.trace(record);
+        logger.trace(record.getMessage());
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.reporting/src/com/tle/core/reporting/ReportingServiceImpl.java:224]:
```suggestion
        logger.trace(message);
```

**Review by edalex-ian — APPROVED** [review]
Wowsers, think I've gone cross-eyed!! But hey, good stuff @PenghaiZhang . Just a couple of minor items, but good to merge when ready.

File: Source/Plugins/Core/com.equella.core/jarsrc/src/main/java/com/tle/web/inplaceeditor/InPlaceEditAppletLauncher.java [hunk 3]
```diff
@@ -523,10 +522,10 @@ public void open() {
 
   private void doOpen() {
     try {
-      logger.entering("InPlaceEditAppletLauncher", "open");
+      logger.trace("InPlaceEditAppletLauncher", "open");
```
[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/jarsrc/src/main/java/com/tle/web/inplaceeditor/InPlaceEditAppletLauncher.java:525]: Yeah I am sure this file is the only one that use `logger.entering`.

[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/jarsrc/src/main/java/com/tle/web/inplaceeditor/InPlaceEditAppletLauncher.java:525]: I was wondering that, too. I could not find `existing` in this file so I thought I might just go with `trace`.

**Review by edalex-ian — APPROVED** [review]
Looks good to merge @PenghaiZhang 

Let me know when it's deployed, then I can also check the logs there.

**Comment by PenghaiZhang** [comment]
> Looks good to merge @PenghaiZhang
> 
> Let me know when it's deployed, then I can also check the logs there.

I just found one more file to change.

**Review by edalex-ian — APPROVED** [review]

**Pull request merged**